cscs-checks/compile/libsci_acc_symlink.py

```diff
@@ -4,21 +4,40 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 import os
+import re
 
 import reframe as rfm
 import reframe.utility.sanity as sn
 
 
+#FIXME: Remove the 71 entries after daint upgrade
 @rfm.parameterized_test(['libsci_acc_gnu_71_nv35'],
                         ['libsci_acc_gnu_71_nv60'],
+                        ['libsci_acc_gnu_81_nv35'],
+                        ['libsci_acc_gnu_81_nv60'],
+                        ['libsci_acc_cray_nv35'],
+                        ['libsci_acc_cray_nv60'],
                         ['libsci_acc_cray_nv35_openacc'],
                         ['libsci_acc_cray_nv60_openacc'])
 class LibSciAccSymLinkTest(rfm.RunOnlyRegressionTest):
     def __init__(self, lib_name):
         self.descr = 'LibSciAcc symlink check of %s' % lib_name
-        self.valid_systems = ['daint:login', 'daint:gpu',
-                              'dom:login', 'dom:gpu',
-                              'tiger:login', 'tiger:gpu']
+
+        self.valid_systems = ['tiger:login', 'tiger:gpu']
+        regex = (r'libsci_acc_(?P<prgenv>[A-Za-z]+)_((?P<cver>[A-Za-z0-9]+)_)'
+                 r'?(?P<version>\S+)')
+        prgenv = re.match(regex, lib_name).group('prgenv')
+
+        #FIXME: Remove this logic after daint upgrade
+        if prgenv == 'gnu':
```

Comment:
Why don’t you simply check the Cray OS and CDT version with the utility functions in os_ext?

Reply:
That would also work. 
Should I use it?

Reply:
The implementation that uses `os_ext.cray_cdt_version()` doesn't change the number of lines, neither is simpler.
Because the logic is inverse. The classes are spawned and then we select if we add or not `dom` or `daint` based on the `lib_name`. While in a "normal" scenario you will be taking action about the environment based on the available CDT, but the systems would be already set.

Reply:
That's fine for me. I just made a suggestion in case you haven't considered.